Code/GraphMol/Matrices.cpp

```diff
@@ -352,20 +346,25 @@ INT_LIST getShortestPath(const ROMol &mol, int aid1, int aid2) {
     boost::tie(nbrIdx, endNbrs) =
         mol.getAtomNeighbors(mol.getAtomWithIdx(curAid));
     while (nbrIdx != endNbrs) {
-      if (doneAtms[*nbrIdx] == 0) {
-        pred[*nbrIdx] = curAid;
-        if (static_cast<int>(*nbrIdx) == aid2) {
+      switch (pred[*nbrIdx]) {
+        case -1:
+          pred[*nbrIdx]=curAid;
+          bfsQ.push_back(rdcast<int>(*nbrIdx));
+          break;
+        case -3: // end found
+          pred[*nbrIdx] = curAid;
           done = true;
+          goto EXIT_LOOP;
+          break;
+        default: // already processed (or begin)
           break;
         }
-        bfsQ.push_back(rdcast<int>(*nbrIdx));
-      }
       nbrIdx++;
```

Comment:
This is bad code that was already there, but it would be great if you could go ahead and change this to `++nbrIdx`